### PR TITLE
Sync Manager for global smart store

### PR DESCRIPTION
**Pull request by kchitalia**

### CODE REVIEW

File: SalesforceSDK/Salesforce.SDK.SmartSync/Manager/SyncManager.cs [hunk 1]
```diff
@@ -50,27 +47,32 @@ public class SyncManager
         public const string LocallyUpdated = "__locally_updated__";
         public const string LocallyDeleted = "__locally_deleted__";
 
-        private static volatile Dictionary<string, SyncManager> _instances;
+        private static volatile Dictionary<string, SyncManager> _instances = new Dictionary<string, SyncManager>();
```
Comment (smcnulty-sfdc) [inline SalesforceSDK/Salesforce.SDK.SmartSync/Manager/SyncManager.cs]: This is set in GetInstance; there are some other pre-setup things that occur if this is not yet set.

File: SalesforceSDK/Salesforce.SDK.SmartSync/Manager/SyncManager.cs [hunk 2]
```diff
@@ -81,17 +83,25 @@ private SyncManager(Account account, string communityId)
...
             }
-            if (account == null)
+
+            if (smartStore == null)
             {
-                return null;
+                smartStore = SmartStore.Store.SmartStore.GetSmartStore(account);
             }
+
             string uniqueId = Constants.GenerateAccountCommunityId(account, communityId);
```
Comment (wmathurin) [inline SalesforceSDK/Salesforce.SDK.SmartSync/Manager/SyncManager.cs]: uniqueId should be computed from smartstore path and user account:
For Android we do:
        String uniqueId = (account != null ? account.getUserId() : "") + ":"
                    + smartStore.getDatabase().getPath();

**Comment by kchitalia** [comment]
@wmathurin

**Comment by wmathurin** [comment]
Looks fine.

**Pull request merged**